docs/polkadot/networking/3-avail-valid.md

```diff
@@ -63,23 +67,24 @@ A4: TODO: Protocols should not be easily spammable, or the spammers should be ea
 
 ### Special considerations
 
-S1: A small fraction of nodes may have poor reachability, and need to communicate via a proxy. Otherwise, we can assume the graph of communication links is mostly-connected via the public internet.
+S1: Some validator nodes are running behind sentry nodes, who must act as their proxy. Otherwise, we can assume that all other nodes (including the sentry nodes themselves) are fully-reachable via the public internet.
 
-This is an optional consideration; we believe we do not need to consider it in the initial implementation, and so our solution for this is described in a separate section in this document. Parts of the main proposal that will need to be changed or extended for this, are marked [TBX S1 #($ref)], TBX standing for "to be extended".
+To help implementation be divided into stages, the main proposal is defined without this consideration. Parts of it that will need to be changed or extended for this, are marked [[TBX S1](#sentry-node-proxies) #($ref)], TBX standing for "to be extended".
 
 ## Protocol overview
 
-We rely on an underlying gossip network that allows us to broadcast various metadata to every node of the relay chain, namely:
+Part of this protocol relies on some pre-existing medium that allows us to broadcast various metadata to every node of the relay chain, namely:
 
 1. receipts of specific pieces ("I have pieces X, Y and Z")
 2. receipts of "enough" pieces ("I have >1/3 of pieces, I don't need more"). This could also double as an attestation to everyone else, that there is availability from the attester's position on the network.
-3. [TBX S1 #1]
 
-This metadata should be gossiped every few seconds. The data of the actual pieces are distributed via a separate topology as described below:
+These should be gossiped every few seconds, and allows the participants to know when the stages of the protocol begin and end, details below.
+
+The data of the actual pieces are distributed via a topology separate from the broadcast medium:
 
 Recall that we have a disjoint partition of N validators into C sets of parachain validators, each set having size N/C. For our purposes for this subprotocol, we will randomly assign a co-ordinate (c, i) to every validator, with c in [0, C) and i in [0, N/C). Fixing c and varying i defines a particular parachain validator set; varying c and fixing i defines what we'll call a particular validator "ring". This name is only meant to be very slightly suggestive, the precise structure and its justification will be described below.
```

Comment:
This is beyond the scope of the current PR, but I wonder about the coordinate scheme in situations where we may have groups of slightly different sizes if `C` is not a divisor of `N`. In practice, we have two options. The first is to spread the remainder over groups, so you have `N%C` groups with 1 extra validator. The other option is to leave some random N%C validators idle for a session. Ensuring that `C` is a divisor poses challenges on implementation and usability so I'd prefer to leave that off the table. Our thinking has been leaning towards option 1 so far, what do you think?

Reply:
Yes, one extra sounds fine.  I still kinda dislike the grid topology here since we can attach fairly regular bipartite graphs every parachain group pair.

Reply:
The grid topology was mainly motivated back when originally I thought I might have to handle nodes with arbitrary NAT situations. So it is not essential, it can be changed. However it does make load-balancing easy to reason about, and I couldn't immediately think of anything that was significantly better.

Reply:
I will add a paragraph mentioning what happens with unevenly-sized parachains, with the "one extra" approach. BTW, this problem occurs with the bipartite-graphs topology as well, since obviously you can't have a perfect matching between sets of size (C) and (C+1).

Reply:
OK, this is actually way less trivial than I thought, at leas to do it in a way that preserves load-balancing properties and avoids some nodes having to do 2x the amount of work. Will have to spend a bit more time to think about this.

Reply:
Note: simply spreading the extra validators across the existing groups, is not well-defined, because you need to define what pieces they are supposed to distribute to everyone else in the group. But there is already someone from their chain, that is distributing a well-defined set of pieces to everyone else in the group.

I will merge this PR and file an issue about this.

Reply:
Tracking in w3f/research-internal#390